crates/rari-doc/src/pages/templates.rs

```diff
@@ -0,0 +1,137 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use super::json::{
+    JsonBlogPostPage, JsonContributorSpotlightPage, JsonCurriculumPage, JsonDocPage,
+    JsonGenericPage, JsonHomePage, JsonSpaPage,
+};
+use super::types::{curriculum, generic};
+
+#[derive(Debug, Clone, Copy, Serialize, Default, Deserialize)]
+pub enum SpaBuildTemplate {
+    #[default]
+    SpaUnknown,
+    SpaNotFound,
+    SpaHomepage,
+    SpaObservatoryLanding,
+    SpaObservatoryAnalyze,
+    SpaAdvertise,
+    SpaPlusLanding,
+    SpaPlusCollections,
+    SpaPlusCollectionsFrequentlyViewed,
+    SpaPlusUpdates,
+    SpaPlusSettings,
+    SpaPlusAiHelp,
+    SpaPlay,
+    SpaSearch,
+}
+
+#[derive(Debug, Clone, Serialize, JsonSchema)]
+#[serde(tag = "renderer")]
+pub enum SpaPage {
```

Comment:
This might move to a macro, but getting the outer API out is more important right now.